### PR TITLE
Add swmm and pyswmm

### PR DESCRIPTION
**Pull request by lrntct**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by lrntct** [comment]
@goanpeca @bemcdonnell I welcome your suggestions.

**Comment by goanpeca** [comment]
LGTM 👍

**Comment by lrntct** [comment]
@conda-forge/staged-recipes, could you please review?

File: recipes/pyswmm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - python
+    - pip
+  run:
+    - six
+    - swmm
+
```
Comment (scopatz) [inline recipes/pyswmm/meta.yaml:26]: This needs a test section

File: recipes/pyswmm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  patches:
+    - __init__.py.patch
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
```
Comment (scopatz) [inline recipes/pyswmm/meta.yaml:20]: These should be in host, not build. Also, does this really not link againt the library?

File: recipes/pyswmm/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/pyswmm/meta.yaml:26]: Oh nm it is below

File: recipes/pyswmm/meta.yaml (hunk 2 above)
Comment (lrntct) [reply recipes/pyswmm/meta.yaml:20]: pyswmm is a Ctypes interface. The C library is not needed until using the Python module.

**Pull request merged**